db/toblock.c

```diff
@@ -2667,6 +2667,31 @@ static int localrep_seqno(tran_type *trans, block_state_t *p_blkstate)
     return rc;
 }
 
+#define MIXED_SQL_DYNTAGS(trans, parent_trans) \
+    { \
+        if (!trans) { \
+            if (osql_needtransaction == OSQL_BPLOG_NOTRANS) { \
+                assert(is_block2sqlmode != 0); \
+                logmsg(LOGMSG_ERROR, "%s:%d INCORRECT TRANSACTION MIX, SQL AND DYNTAG\n", \
+                        __FILE__, __LINE__); \
+                rc = osql_create_transaction( \
+                        javasp_trans_handle, iq, &trans, \
```

Comment:
in case it is not clear, trans is first, parent_trans is second. That is the bug.